### PR TITLE
Changed the name of the node visiting function from `visit_Num` to `visit_Constant`.

### LINKED ISSUE #23092: ast_parser: PendingDeprecationWarning: visit_Num is deprecated; add visit_Constant
`sympy.parsing.ast_parser` triggers `PendingDeprecationWarning` in Python ≥ 3.8:
```console
$ python3 -Wd
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy.parsing.ast_parser import parse_expr
>>> parse_expr('6 * 7', {})
/usr/lib/python3.8/ast.py:371: PendingDeprecationWarning: visit_Num is deprecated; add visit_Constant
  return visitor(node)
42
```

### PR DESCRIPTION
**Pull request by wermos**
#### References to other Issues or PRs

Fixes #23092 

#### Brief description of what is fixed or changed
I changed the name of the node visiting function `visit_Num` to `visit_Constant` because the former is deprecated since version 3.8.

#### Release Notes

* parsing
  * Changed the name of the `visit_Num` node visitor function to `visit_Constant`, as the former is deprecated.

### CODE REVIEW

**Comment by wermos** [comment]
Hi, this is my first PR, and it appears that the sympy-bot wants me to add my name and email to the `.mailmap` file. Should I go ahead and do that?

**Comment by anutosh491** [comment]
> Hi, this is my first PR, and it appears that the sympy-bot wants me to add my name and email to the `.mailmap` file. Should I go ahead and do that?

Well I guess you could have added your name and github account to the mailmap file in the pr itself .Opening a separate Pr for that wasn't necessary!

**Comment by wermos** [comment]
@anutosh491 The other PR is specifically for fixing a grammatical mistake in the HELP section of the `.mailmap` file. As for adding myself, I will do it in this PR itself.

Thank you for your feedback! 😁

**Comment by wermos** [comment]
I did my best to follow the given instructions. I hope I didn't mess anything up 😅

**Comment by oscarbenjamin** [comment]
The instructions in the .mailmap file itself are out of date. What you need to do is run the `bin/mailmap_check.py` script to reorder the lines in the .mailmap file.

**Comment by oscarbenjamin** [comment]
You don't need to close this PR. Just commit the changes on the same branch and push to update the PR.

File: AUTHORS [hunk 1]
```diff
@@ -1133,3 +1133,5 @@ Tom Fryers <61272761+TomFryers@users.noreply.github.com>
 Zouhair <zouhair.mahboubi@gmail.com>
 zzj <29055749+zjzh@users.noreply.github.com>
 shubhayu09 <guptashubhayu601@gmail.com>
+Siddhant Jain <siddhantashoknagar@gmail.com>
+Tirthankar Mazumder <63574588+wermos@users.noreply.github.com>
```
Comment (oscarbenjamin) [inline AUTHORS:1137]: The changes to the AUTHORS file should be reverted. The AUTHORS file will be updated automatically at the time of the next release based on the information in the .mailmap file.

File: AUTHORS [hunk 2]
```diff
@@ -4,7 +4,7 @@ those who explicitly didn't want to be mentioned. People with a * next
 to their names are not found in the metadata of the git history. This
 file is generated automatically by running `./bin/authors_update.py`.
 
-There are a total of 1127 authors.
+There are a total of 1129 authors.
```
Comment (oscarbenjamin) [inline AUTHORS:7]: This change should also be reverted.

File: sympy/parsing/ast_parser.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def __init__(self, local_dict, global_dict):
         self.local_dict = local_dict
         self.global_dict = global_dict
 
-    def visit_Num(self, node):
+    def visit_Constant(self, node):
```
[the remark this answers is not in the document]
Comment (oscarbenjamin) [reply sympy/parsing/ast_parser.py:35]: Also tests should be added for all of these cases.

**Pull request merged**